forge/ee/db/controllers/Pipeline.js

```diff
@@ -432,6 +432,9 @@ module.exports = {
                 // Update the targetSnapshot of the device group
                 await targetDeviceGroup.PipelineStageDeviceGroup.update({ targetSnapshotId: sourceSnapshot.id }, { transaction })
 
+                // Update the targetSnapshotId on the device group
+                await targetDeviceGroup.update({ targetSnapshotId: sourceSnapshot.id }, { transaction })
+
```

Comment:
A possibly cleaner approach would be to have a controller for DeviceGroup which provides a function to update the targetSnapshotId which takes care of updating the devices *and* sending the update command. As it is we've got that logic spread through this Pipeline controller. Not blocking, just a thought.

Reply:
I _could_ add a method to DeviceGroup controller but I dont want to be updating devices until the transaction is committed.